crates/core/src/live_socket/socket.rs

```diff
@@ -10,6 +11,29 @@ use crate::{
     parser::parse,
 };
 
+#[derive(Debug, Clone, PartialEq, Eq, uniffi::Record)]
+pub struct ConnectOpts {
+    #[uniffi(default = None)]
+    pub headers: Option<HashMap<String, String>>,
+    #[uniffi(default = None)]
+    pub body: Option<String>,
+    #[uniffi(default = "GET")]
+    pub method: String,
+    #[uniffi(default = 30_000)]
+    pub timeout_ms: u64,
+}
+
+impl Default for ConnectOpts {
+    fn default() -> Self {
+        Self {
+            headers: None,
+            body: None,
+            method: String::from("GET"),
+            timeout_ms: 30_000,
```

Comment:
I'm on the fence about this being a magic number or a `const LIVE_SOCKET_TIMEOUT` defined someplace.